### PR TITLE
obs-websocket: init at 4.9.0

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Motivation for this change

OBS websocket plugin enables users to remote control their OBS-Studio instance over the network.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2927,6 +2927,16 @@
     githubId = 147284;
     name = "Jason Felice";
   };
+  erdnaxe = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2930]: Please put this into a separate commit with the message ``maintainers: add erdnaxe`` and sort it alphabetically.

File: pkgs/applications/video/obs-studio/obs-websocket.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  in ''
+    mkdir -p $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
+    ln -s $out/lib/obs-plugins/obs-websocket.so $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
+  '';
+
+  meta = with lib; {
+    description = "Remote-control OBS Studio through WebSockets";
+    homepage = "https://github.com/Palakis/obs-websocket";
+    maintainers = with maintainers; [ erdnaxe ];
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/obs-websocket.nix:48]: gpl2Only or gpl2Plus?

File: pkgs/applications/video/obs-studio/obs-websocket.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  # obs-studio expects the shared object to be located in bin/32bit or bin/64bit
+  # https://github.com/obsproject/obs-studio/blob/d60c736cb0ec0491013293c8a483d3a6573165cb/libobs/obs-nix.c#L48
+  postInstall = let
+    pluginPath = {
+      i686-linux = "bin/32bit";
+      x86_64-linux = "bin/64bit";
+    }.${stdenv.targetPlatform.system} or (throw "Unsupported system: ${stdenv.targetPlatform.system}");
+  in ''
+    mkdir -p $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
+    ln -s $out/lib/obs-plugins/obs-websocket.so $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/obs-websocket.nix:41]: Maybe move it if it is only required in the target directory.

File: pkgs/applications/video/obs-studio/obs-websocket.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    sha256 = "1dxih5czcfs1vczbq48784jvmgs8awbsrwk8mdfi4pg8n577cr1w";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ qtbase obs-studio asio_1_10 websocketpp ];
+
+  dontWrapQtApps = true;
+
+  cmakeFlags = with lib; [
+    "-DLIBOBS_INCLUDE_DIR=${obs-studio.src}/libobs"
```
Comment (SuperSandro2000) [inline pkgs/applications/video/obs-studio/obs-websocket.nix:29]:
```suggestion
  cmakeFlags = [
    "-DLIBOBS_INCLUDE_DIR=${obs-studio.src}/libobs"
```

File: pkgs/applications/video/obs-studio/obs-websocket.nix (hunk 2 above)
Comment (erdnaxe) [reply pkgs/applications/video/obs-studio/obs-websocket.nix:48]: From the LICENSE:
>    This program is free software; you can redistribute it and/or modify
>    it under the terms of the GNU General Public License as published by
>    the Free Software Foundation; either version 2 of the License, or
>    (at your option) any later version.

So I guess it's gpl2Plus?

File: pkgs/applications/video/obs-studio/obs-websocket.nix (hunk 3 above)
Comment (erdnaxe) [reply pkgs/applications/video/obs-studio/obs-websocket.nix:41]: It seems to be required. It is also done by <https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/video/obs-studio/v4l2sink.nix#L38> and <https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/video/obs-studio/obs-gstreamer.nix#L25>.

File: pkgs/applications/video/obs-studio/obs-websocket.nix (hunk 2 above)
Comment (lukegb) [reply pkgs/applications/video/obs-studio/obs-websocket.nix]: Right; the "at your option any later version" clause means it's gpl2Plus.

File: pkgs/applications/video/obs-studio/obs-websocket.nix (hunk 3 above)
Comment (erdnaxe) [reply pkgs/applications/video/obs-studio/obs-websocket.nix:41]: Should I contact obs-studio Nix package maintainer to have the idea clear on that?

Comment (deviant) [reply pkgs/applications/video/obs-studio/obs-websocket.nix:41]: This won't be necessary once #125308 is merged. :)

Comment (erdnaxe) [reply pkgs/applications/video/obs-studio/obs-websocket.nix]: Thanks, I will add the plugin to `pkgs/applications/video/obs-studio/plugins/default.nix` once the plugin wrapper is merged :)
